src/testnet/node.rs

```diff
@@ -0,0 +1,394 @@
+use chainstate::burn::operations::{BlockstackOperationType, LeaderKeyRegisterOp, LeaderBlockCommitOp};
+use chainstate::burn::{ConsensusHash, SortitionHash, BlockSnapshot, VRFSeed, BlockHeaderHash};
+use net::StacksMessageType;
+use util::hash::Sha256Sum;
+use util::vrf::{VRFProof, VRFPublicKey};
+
+const TESTNET_CHAIN_ID: u32 = 0x80000000;
+
+#[derive(Clone)]
+struct RegisteredKey {
+    block_height: u16,
+    op_vtxindex: u16,
+    vrf_public_key: VRFPublicKey,
+}
+
+#[derive(Clone, Debug)]
+pub struct SortitionedBlock {
+    pub block_height: u16,
+    pub burn_header_hash: BurnchainHeaderHash,
+    consensus_hash: ConsensusHash,
+    op_vtxindex: u16,
+    op_txid: Txid,
+    parent_burn_header_hash: BurnchainHeaderHash,
+    sortition_hash: SortitionHash,
+    total_burn: u64,
+}
+
+impl SortitionedBlock {
+    pub fn genesis() -> Self {
+        Self {
+            sortition_hash: SortitionHash::initial(),
+            consensus_hash: ConsensusHash::empty(),
+            burn_header_hash: BurnchainHeaderHash([0u8; 32]),
+            parent_burn_header_hash: BurnchainHeaderHash([0u8; 32]),
+            block_height: 1,
+            op_vtxindex: 0,
+            op_txid: Txid([0u8; 32]),
+            total_burn: 0
+        }
+    }
+
+}
+
+/// 
+pub struct Node {
+    active_registered_key: Option<RegisteredKey>,
+    average_block_time: u64,
+    bootstraping_chain: bool,
+    burnchain_ops_tx: Option<Sender<BlockstackOperationType>>,
+    burnchain_tip: Option<BlockSnapshot>,
+    chain_state: StacksChainState,
+    chain_tip: Option<StacksHeaderInfo>,
+    config: NodeConfig,
+    keychain: Keychain,
+    last_sortitioned_block: Option<SortitionedBlock>,
+    mem_pool: MemPoolFS,
+    nonce: u64,
+    pub rx: Receiver<StacksMessageType>,
+    pub tx: Sender<StacksMessageType>,
+}
+
+impl Node {
+
+    /// 
+    pub fn new(config: NodeConfig, average_block_time: u64) -> Self {
+        
+        let keychain = Keychain::default();
+
+        let chain_state = match StacksChainState::open(false, 0x80000000, &config.path) {
+            Ok(res) => res,
+            Err(err) => panic!("Error while opening chain state at path {:?}", config.path)
+        };
+
+        let mem_pool = MemPoolFS::new(&config.mem_pool_path);
+
+        let (tx, rx) = channel();
+
+        Self {
+            active_registered_key: None,
+            bootstraping_chain: false,
+            chain_state,
+            chain_tip: None,
+            config,
+            keychain,
+            last_sortitioned_block: None,
+            mem_pool,
+            average_block_time,
+            burnchain_ops_tx: None,
+            burnchain_tip: None,
+            tx,
+            rx,
+            nonce: 0,
+        }
+    }
+    
+    /// 
+    pub fn tear_up(&mut self, burnchain_ops_tx: Sender<BlockstackOperationType>) {
```

Comment:
Not a bug, but can we change this method name to `setup` or something?  `tear_up` is synonymous with destroying something (e.g. "to tear up a piece of paper" means to rip it into lots of tiny pieces).

Reply:
Ha ha, ok my bad. addressed with https://github.com/blockstack/blockstack-core/pull/1202/commits/4c10caa9c59614cad176b88ac92982eff3634b20